tests/e2e/util.go

```diff
@@ -2684,6 +2689,21 @@ func toggleCSIMigrationFeatureGatesOnK8snodes(ctx context.Context, client client
 	}
 }
 
+//isCSIMigrationFeatureGatesEnabledOnKubelet checks whether CSIMigration Feature Gates are enabled on CSI Node
+func isCSIMigrationFeatureGatesEnabledOnKubelet(ctx context.Context, client clientset.Interface, nodeName string) bool {
+	csinode, err := client.StorageV1().CSINodes().Get(ctx, nodeName, metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	found := false
+	for annotation, value := range csinode.Annotations {
+		framework.Logf("Annotation seen on CSI node - %s:%s", annotation, value)
+		if annotation == migratedPluginAnnotation && strings.Contains(value, vcpProvisionerName) {
+			found = true
+			break
+		}
+	}
+	return found
+}
```

Comment:
the function will return true even if some of the CSI nodes are not enabled with CSIMigration Feature Gates
need a else - which resets found to false

Reply:
This will not loop through csinodes, this will only run for one node.